Extension/src/LanguageServer/configurations.ts

```diff
@@ -739,29 +740,47 @@ export class CppProperties {
         const result: string[] = [];
         if (paths) {
             paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
-        if (!glob) {
-            return result;
+            if (paths && !glob) {
+                paths.forEach(entry => {
+                    const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
```

Comment:
util.resolveVariables isn't getting called if glob is true

Reply:
that is what we want because as far as I can tell, resolve variables is just for resolving the path delimiters

Reply:
Doesn't it resolve variables?

Reply:
Oh I see I totally forgot about that functionality. Thanks for pointing it out